### PR TITLE
Managers can add notes and set status of reports

### PR DESCRIPTION
**Pull request by jasalisbury**
**Description of change**
 * Review page is now different if the user looking at the page is the approving manager
 * Manager's review page includes the user's additional note and allows the manager to leave a manager note
 * Manager note added to DB
 * Report state is shown in the side nav's "review" section once it leaves "draft" mode
 * Policy, service and route added to update status of a report

**How to test**
1) Pull down, run migrations/seeders
2) Make sure the `CURRENT_USER_ID` is set to "1"
3) Fill out a complete report. Add "Hermione" as the approving manager
4) Refresh the report, since Hermione is user with the id 1 you should see the approving manager version of the review page.
5) Add a manager note and set the status. Note once you submit the status on the side nav `Review and submit` section updates
6) Refresh the page to verify the manager notes persists

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/138
* https://github.com/HHS/Head-Start-TTADP/issues/231
* https://github.com/HHS/Head-Start-TTADP/issues/236
* https://github.com/HHS/Head-Start-TTADP/issues/137

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

**Comment by jasalisbury** [comment]
API docs aren't working for me for some reason, will look into it.

**Comment by dcmcand** [comment]
Currently the manager notes are not connected to a particular user. I know that there will only be one approving manager per activity report, but I could see a situation where the approving manager gets changed. Is this a concern? Something to look at post mvp?

**Comment by jasalisbury** [comment]
> Currently the manager notes are not connected to a particular user. I know that there will only be one approving manager per activity report, but I could see a situation where the approving manager gets changed. Is this a concern? Something to look at post mvp?

That is a good question. My first thought is to not worry about it until after the MVP, thoughts @rahearn?

**Comment by rahearn** [comment]
> My first thought is to not worry about it until after the MVP

Yeah, it's not ideal but I think we can punt this until we implement multiple approvals.

File: frontend/src/pages/ActivityReport/Pages/ReviewSubmit.js
Comment (gopar) [inline frontend/src/pages/ActivityReport/Pages/ReviewSubmit.js:78]: How do people feel about moving these 2 main sections into their own components? (`!approvingManager` and `approvingManager`)

File: frontend/src/fetchers/activityReports.js
Comment (gopar) [inline frontend/src/fetchers/activityReports.js:44]: Would doing this improve readability?
```javascript
const DECIMAL_BASE = 10
....
reportId.toString(DECIMAL_BASE)
```

I only mention this b/c of magic numbers, and I had to look up what the extra param in `toString` was. This is also due to me not being familiar with JS that much, so I'll defer to you JS vets if it makes sense to change :)

File: docs/openapi/paths/activity-reports/review.yaml
Comment (gopar) [inline docs/openapi/paths/activity-reports/review.yaml:18]: Should we include an enum since we know there's a set of values that will be accepted?

File: frontend/src/pages/ActivityReport/Pages/ReviewSubmit.js
Comment (jasalisbury) [reply frontend/src/pages/ActivityReport/Pages/ReviewSubmit.js:78]: I originally split them out but decided against (for no real reason). I can split them out again if you think that will help with readability

File: frontend/src/fetchers/activityReports.js
Comment (jasalisbury) [reply frontend/src/fetchers/activityReports.js:44]: > Would doing this improve readability?

:+1: Yes it would

**Review by dcmcand — APPROVED** [review]
Looks good

**Review by gopar — APPROVED** [review]

**Pull request merged**
